cruise.umple/src/UmpleInternalParser_CodeParserHandlers.ump

```diff
@@ -29,39 +37,69 @@ class UseStatementParserAction{
   depend cruise.umple.parser.Token;
   
   isA ParserAction;
-  
+  UmpleModel umpleModel;
+ 
   public void onSuccess(Token token, ParserDataPackage data)
   {
+  /*
+  we can filter the use statement by checking if ends with .ump 
+  this is simpler and needs few modification
+  
+  the solution 2: if no file, save the mixset name for later use .
+  this can be done by do some changes to the file .ump directly. Or, we can make the file throw a null pointer exception and hendel it in UseStatementAction
+  in .ump
+  
+  */
+ 
     String value = token.getValue("use");
     String fileName="";
-		String path = "";    
+	String path = "";    
     int index =-1;
     if (data!=null && data.getFullFileAddress()!=null)    index = data.getFullFileAddress().lastIndexOf("/") ;
     if (index==-1) index = data.getFullFileAddress().lastIndexOf("\\");  
     if (index!=-1){
     	path = data.getFullFileAddress().substring(0,index+1 );
     }
     
-    synchronized(data.getHasParsed())
+synchronized(data.getHasParsed())
     {
       fileName = data.getAnalyzer().getFile().getAbsoluteFile().getParentFile().getAbsolutePath() + File.separator + value;
       //I used this technique because I noticed this function is used to several purposes.
       // The correct implementation is just to set fileName = path +value; and remove the above code.
       //TODO: this issue must be investigated more. Currently it is working with the following solution. 
       File f = new File(fileName);
       if (!f.exists()) fileName = path +value;
-      if(!data.getHasParsed().contains(f.getAbsolutePath()))
+     
+     /*  
+      *	when a file is not exist, consider it as a possible mixset. 
```

Comment:
I would say definitively: Any thing ending with '.ump' must be a file. Anything not ending with .ump must be a mixset.

Reply:
This will make the work much easy.